src/Sulu/Bundle/ContentBundle/Tests/Unit/Repository/ResourceLocatorRepositoryTest.php

```diff
@@ -0,0 +1,225 @@
+        ]);
+        $this->structureManager->getStructure($template)->willReturn($structure->reveal());
+
+        $grandParent = $this->prophesize(BasePageDocument::class);
+        $grandParent->getPublished()->willReturn(true);
+
+        $parent = $this->prophesize(PageDocument::class);
+        $parent->getPublished()->willReturn(false);
+        $parent->getParent()->willReturn($grandParent->reveal());
+
+        $this->documentManager->find($parentUuid, $locale, ['load_ghost_content' => false])->willReturn($parent->reveal());
+
+        $grandParentPath = '/grandparent';
+        $this->rlpStrategy->loadByContent($grandParent)->willReturn($grandParentPath);
+
+        $title = 'news-football';
+        $path = '/grandparent/news-football';
+        $this->rlpStrategy->generate($title, $grandParentPath, $webspace, $locale, null)->willReturn($path);
+
+        $result = $this->repository->generate($parts, $parentUuid, null, $webspace, $locale, $template);
+        $this->assertEquals($result['resourceLocator'], $path);
+    }
+
+    public function testGenerateWithUuid()
+    {
+        $parts = [
+            'title' => 'news',
+            'subtitle' => 'football',
+        ];
+        $uuid = '0123456789abcdef';
+        $webspace = 'sulu_io';
+        $locale = 'en';
+        $template = 'default';
+
+        $structure = $this->prophesize(StructureInterface::class);
+        $structure->getPropertiesByTagName('sulu.rlp.part')->willReturn([
+            new Property('subtitle', 'subtitle', 'subtitle'),
+            new Property('title', 'title', 'title'),
+        ]);
+        $this->structureManager->getStructure($template)->willReturn($structure->reveal());
+
+        $title = 'news-football';
+        $path = '/ancestor/news-football';
+        $this->rlpStrategy->generateForUuid($title, $uuid, $webspace, $locale, null)->willReturn($path);
+
+        $result = $this->repository->generate($parts, null, $uuid, $webspace, $locale, $template);
+        $this->assertEquals($result['resourceLocator'], $path);
+    }
+
+    public function testGenerate()
+    {
+        $parts = [
+            'title' => 'news',
+            'subtitle' => 'football',
+        ];
+        $webspace = 'sulu_io';
+        $locale = 'en';
+        $template = 'default';
+
+        $structure = $this->prophesize(StructureInterface::class);
+        $structure->getPropertiesByTagName('sulu.rlp.part')->willReturn([
+            new Property('subtitle', 'subtitle', 'subtitle'),
+            new Property('title', 'title', 'title'),
+        ]);
+        $this->structureManager->getStructure($template)->willReturn($structure->reveal());
+
+        $title = 'news-football';
+        $path = '/news-football';
+        $this->rlpStrategy->generate($title, '/', $webspace, $locale, null)->willReturn($path);
+
+        $result = $this->repository->generate($parts, null, null, $webspace, $locale, $template);
+        $this->assertEquals($result['resourceLocator'], $path);
```

Comment:
Well, it is not really about intuition here. It could be possible that the variable is modified somewhere in between... But yeah, it's not the case here, forget about it.